### ISSUE TITLE
Update GoogleTest version

### ISSUE THREAD

**Issue by amitdo** [tesseract-ocr/tesseract#3633]
Currently we use 1.10.0 which is ~2 years old.

Maybe we can upgrade to 1.11.0.

https://github.com/google/googletest/tree/B4DE7B5F4F45390ABCFB887AFADA9303

**Comment by stweil** [comment]
Newer versions make trouble for builds with sanitizers (unresolved symbol when linking).

**Comment by amitdo** [comment]
Can we solve it in our side, or is this a bug that needs to be reported to GoogleTest devs?

It can be postponed to post 5.0.0 release.

**Comment by CBaiz** [comment]
@stweil 
Which linker errors occur exactly?
I had a project which also produced linker errors when upgrading to 1.11.0 but it was easily solved by removing GTEST_API_ in front of all my main functions calling RUN_ALL_TESTS.
My error(s) were: Compiler Error C2491: 'identifier' : definition of dllimport function not allowed
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2491?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(C2491);%26rd%3Dtrue&view=msvc-170

Before: GTEST_API_ int main(int argc, char **argv)
After: int main(int argc, char **argv)

OS: Windows
Compiler: MSVC (14.1 = Visual Studio 2017)

**Comment by stweil** [comment]
I cannot reproduce the linker error and therefore now decided to upgrade GoogleTest, see commit a185c068d1ace62d91285d094a3858135cb0cefe.
